### PR TITLE
Add a pattern to convert mhlo.is_finite -> linalg.generic

### PR DESCRIPTION
**Pull request by asaadaldien**
Convert `mhlo.is_finite` with float32 to pointwise `linalg.generic`comparing.
This should work for all targets implementing IEEE 754 float arithmetic.

### CODE REVIEW

File: iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp [hunk 1]
```diff
@@ -42,15 +42,53 @@ namespace mlir {
 namespace iree_compiler {
 namespace {
 
+// Convert mhlo.is_finite to linalg.generic for float32 values ranked inputs.
```
Comment (MaheshRavishankar) [inline iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp:45]: THis doesnt need to be here. Could you add this to TF MHLO -> Linalg conversion.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp (hunk 1 above)
Comment (asaadaldien) [reply iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp:45]: This is going to be correct only for HW implementing IEEE 754 float arithmetic where f32 inf value = ` 0x7F800000` that was my concern regarding having this in TF.

Comment (MaheshRavishankar) [reply iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp:45]: I think TF uses values like PI etc.  I would guess they also have some place they use for INF. Could we check with TF folks. If they dont, then we can put this in IREE

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]

File: iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp (hunk 1 above)
Comment (asaadaldien) [reply iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp:45]: SGTM! I will get in touch with TF folks.

**Comment by asaadaldien** [comment]
Submitting to TF instead

**Pull request closed without merging**